src/GradhSph/GradhSph.cpp

```diff
@@ -261,7 +261,7 @@ int GradhSph<ndim, kernelclass>::ComputeH
 
     // Density must at least equal its self-contribution
     // (failure could indicate neighbour list problem)
-    assert(parti.rho >= parti.m*parti.hfactor*kern.w0(0.0));
+    assert(parti.rho >= (FLOAT) 0.999*parti.m*parti.hfactor*kern.w0(0.0));
```

Comment:
There's no need for the 0.999 factor. Also, the assert should be using the w0_s2 kernel rather than w0. I think I have already corrected it already in some other branch...